includes/global.sh

```diff
@@ -126,39 +126,48 @@ echo "Jail creation completed for ${1}"
 }
 
 initblueprint() {
-blueprint=jail_${1}_blueprint
-varlist=blueprint_${!blueprint}_vars
+	blueprint=jail_${1}_blueprint
+	varlist=blueprint_${!blueprint}_vars
+
+	for var in ${!varlist} ${global_jails_vars}
+	do
+		value="jail_${1}_$var"
+		val=${!value:-}
+		declare -g "${var}=${val}"
+
+		if [[ "${var}" =~ ^link_.* ]]; 
+		then
+			linkblueprint=jail_${val}_blueprint
+			linkvarlist=blueprint_${!linkblueprint}_vars
+			for linkvar in ${!linkvarlist} ${global_jails_vars}
+			do
+				linkvalue="jail_${val}_${linkvar}"
+				linkval=${!linkvalue:-}
+				declare -g "${var}_${linkvar}=${linkval}"
+			done
+		fi
+	done
 
-for var in ${!varlist} ${global_jails_vars}
-do
-	value="jail_${1}_$var"
-	val=${!value:-}
-	declare -g "${var}=${val}"
-	
-	if [[ "${var}" =~ ^link_.* ]]; 
-	then
-		linkblueprint=jail_${val}_blueprint
-		linkvarlist=blueprint_${!linkblueprint}_vars
-		for linkvar in ${!linkvarlist} ${global_jails_vars}
-		do
-			linkvalue="jail_${val}_${linkvar}"
-			linkval=${!linkvalue:-}
-			declare -g "${var}_${linkvar}=${linkval}"
-		done
-	fi
-done
+	declare -g "includes_dir=${SCRIPT_DIR}/blueprints/${!blueprint}/includes"
 
-declare -g "includes_dir=${SCRIPT_DIR}/blueprints/${!blueprint}/includes"
+	if [ -f "/mnt/${global_dataset_config}/${1}/INSTALLED" ]; then
+	    echo "Reinstall detected..."
+		declare -g reinstall="true"
+	elif [ "$(ls -A "/mnt/${global_dataset_config}/${1}/")" ]; then
+	    echo "ERROR, No valid install detected in config directory but files present"
+		exit 1
+	else
+		echo "No reinstall flag detected, continuing normal install"
+	fi
 
-if [ -f "/mnt/${global_dataset_config}/${1}/INSTALLED" ]; then
-    echo "Reinstall detected..."
-	declare -g reinstall="true"
-elif [ "$(ls -A "/mnt/${global_dataset_config}/${1}/")" ]; then
-    echo "ERROR, No valid install detected in config directory but files present"
-	exit 1
-else
-	echo "No reinstall flag detected, continuing normal install"
-fi
+	jailip4="jail_${1}_ip4_addr"
+	ip4val=${!jailip4:-}
+	if [ -z "${ip4val}" ]; then
+		DEFAULT_IF=$(iocage exec "$1" route get default | awk '/interface/ {print $2}')
+		declare -g "jail_ip=$(iocage exec "$1" ifconfig "$DEFAULT_IF" | awk '/inet/ { print $2 }')"
+	else
+		declare -g "jail_ip=${ip4_addr%/*}"
+	fi
```

Comment:
Some suggestions, ip4_addr is already defined at this stage...

```suggestion
	if [ -z "${ip4_addr}" ]; then
		DEFAULT_IF=$(iocage exec "$1" route get default | awk '/interface/ {print $2}')
		declare -g "jail_ip=$(iocage exec "$1" ifconfig "$DEFAULT_IF" | awk '/inet/ { print $2 }')"
	else
		declare -g "jail_ip=${ip4_addr%/*}"
	fi
```

Reply:
ip4_addr is undefined the whole time if using dhcp, that's why i have the extra check. so this works with both dhcp and a defined ip for the jail.

Reply:
No it is defined as "".
The for loop runs ALL variables that are listed under "var" or the global "var".
Look:
```
for var in ${!varlist} ${global_jails_vars}
do
	value="jail_${1}_$var"
	val=${!value:-}
	declare -g "${var}=${val}"
```

Reply:
It's part of "global_jails_vars", so it is being processed regardless of being set in config.yml.
Thats why your script errored out without `val=${!value:-}`

Reply:
Ok, clarification: `-z "$var"` tests for undefined OR empty, which is the case when using dhcp.

Reply:
So what is the problem with my solution, did you test it and it didn't work?

Reply:
Ah, yes I think it ran into an issue, but I’d need to test again to refresh my memory.

Reply:
Can't repro, simplified it!